src/components/toolbox/switcher/index.js

```diff
@@ -1,22 +1,36 @@
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { useRef } from 'react';
+import { withRouter } from 'react-router';
+import { addSearchParamsToUrl } from '../../../utils/searchParams';
 import styles from './switcher.css';
 
-const Switcher = ({ options, onClick, active }) => (
+const TabLink = withRouter(({
+  children, to, className, history, data,
+}) => {
+  const linkEl = useRef(null);
+  const onClick = () => {
+    addSearchParamsToUrl(history, { tab: to, ...data });
+  };
+
+  return (
+    <li onClick={onClick} ref={linkEl} className={className} data-value={to}>{ children }</li>
+  );
+});
+
+const Switcher = ({ options, active }) => (
   <div className={styles.wrapper}>
     <ul className={styles.options}>
       {options.map(tab => (
-        <li
+        <TabLink
           className={[
             tab.value === active && styles.active,
             tab.className,
           ].filter(Boolean).join(' ')}
           key={tab.value}
-          data-value={tab.value}
-          onClick={onClick}
+          to={tab.value}
```

Comment:
It would be better if we use an `id` to identify the tab instead of the tab name (tab.value). 
With the above implementation, the routes will look like: `...?tab=Delegate%20profile`
Instead, it would be better to have: `...?tab=delegateProfile`